src/NHibernate/Type/GenericSetType.cs

```diff
@@ -76,5 +78,15 @@ public override object Instantiate(int anticipatedSize)
 		{
 			return new HashSet<T>();
 		}
+
+		protected override bool AreCollectionElementsEqual(IEnumerable original, IEnumerable target)
```

Comment:
This can be just `return ((ISet<T>) original).SetEquals((IEnumerable<T>) target);`